docs/migration/index.md

```diff
@@ -0,0 +1,382 @@
+---
+order: 1
+title: v0.18
+parent:
+  title: Migration
+  order: 3
+description: For chains that were scaffolded with Starport versions lower than v0.18, changes are required to use Starport v0.18. 
+---
+
+# Upgrading a Blockchain to use Starport v0.18
+
+Starport v0.18 comes with Cosmos SDK v0.44. This version of Cosmos SDK introduced changes that are not compatible with chains that were scaffolded with Starport versions lower than v0.18.
+
+**Important:** After upgrading from Starport v0.17.3 to Starport v0.18, you must update the default blockchain template to use blockchains that were scaffolded with earlier versions.
+
+These instructions are written for a blockchain that was scaffolded with the following command:
+
+```
+starport scaffold chain github.com/cosmonaut/mars
+```
+
+If you used a different module path, replace `cosmonaut` and `mars` with the correct values for your blockchain.
+
+## Blockchain
+
+For each file listed, make the required changes to the source code of the blockchain template. 
+
+### app/app.go
+
+```go
+import (
+  //...
+  // Add the following packages:
+  "github.com/cosmos/cosmos-sdk/x/feegrant"
+  feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
+  feegrantmodule "github.com/cosmos/cosmos-sdk/x/feegrant/module"
+
+  "github.com/cosmos/ibc-go/modules/apps/transfer"
+  ibctransferkeeper "github.com/cosmos/ibc-go/modules/apps/transfer/keeper"
+  ibctransfertypes "github.com/cosmos/ibc-go/modules/apps/transfer/types"
+  ibc "github.com/cosmos/ibc-go/modules/core"
+  ibcclient "github.com/cosmos/ibc-go/modules/core/02-client"
+  ibcporttypes "github.com/cosmos/ibc-go/modules/core/05-port/types"
+  ibchost "github.com/cosmos/ibc-go/modules/core/24-host"
+  ibckeeper "github.com/cosmos/ibc-go/modules/core/keeper"
+  
+  // Remove the following packages:
+  // transfer "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer"
+  // ibctransferkeeper "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/keeper"
+  // ibctransfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+  // ibc "github.com/cosmos/cosmos-sdk/x/ibc/core"
+  // ibcclient "github.com/cosmos/cosmos-sdk/x/ibc/core/02-client"
+  // porttypes "github.com/cosmos/cosmos-sdk/x/ibc/core/05-port/types"
+  // ibchost "github.com/cosmos/cosmos-sdk/x/ibc/core/24-host"
+  // ibckeeper "github.com/cosmos/cosmos-sdk/x/ibc/core/keeper"
+)
+
+var (
+  //...
+  ModuleBasics = module.NewBasicManager(
+    //...
+    slashing.AppModuleBasic{},
+    // Add feegrantmodule.AppModuleBasic{},
+    feegrantmodule.AppModuleBasic{}, // <--
+    ibc.AppModuleBasic{},
+    //...
+  )
+  //...
+)
+
+type App struct {
+  //...
+  // Replace codec.Marshaler with codec.Codec
+  appCodec          codec.Codec // <--
+  // Add FeeGrantKeeper
+  FeeGrantKeeper   feegrantkeeper.Keeper // <--
+}
+
+``go
```

Comment:
This is wrong. Either remove it, have an ending statement for the last part. Also, 3 instead of two `

Reply:
If you meant just the double-tick go part here, then ok.

Reply:
Yes